aspnet/fundamentals/servers.rst

```diff
@@ -95,77 +100,54 @@ ASP.NET defines a number of :doc:`request-features`. The following table lists t
 
 .. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/Startup.cs
   :linenos:
-  :lines: 37-54
-  :emphasize-lines: 3,6-7,10,15
-  :language: c#
+  :lines: 25-46
+  :emphasize-lines: 5
+  :language: none
   :dedent: 8
 
+.. note:: Binding to ``http://localhost:0`` is not supported. You must either bind to ``http://127.0.0.1:0``, ``http://[::1]:0`` or both individually.
 
 IIS and IIS Express
 -------------------
 
 IIS is the most feature rich server, and includes IIS management functionality and access to other IIS modules. Hosting ASP.NET Core no longer uses the ``System.Web`` infrastructure used by prior versions of ASP.NET.
 
-HTTPPlatformHandler
-^^^^^^^^^^^^^^^^^^^
-In ASP.NET Core on Windows, the web application is hosted by an external process outside of IIS. The HTTP Platform Handler is an IIS 7.5+ module which is responsible for process management of HTTP listeners and used to proxy requests to the processes that it manages.
+IIS Express can be launched by Visual Studio using the default profile defined by the ASP.NET Core templates. :ref:`publishing-and-deployment` provides guidelines for publishing to IIS.
 
-.. _weblistener:
+ASP.NET Core Module
+^^^^^^^^^^^^^^^^^^^
+In ASP.NET Core on Windows, the web application is hosted by an external process outside of IIS. The ASP.NET Core Module is an IIS 7.5+ module which is responsible for process management of HTTP listeners and used to proxy requests to the processes that it manages.
 
-WebListener
------------
+.. _kestrel:
 
-WebListener is a Windows-only HTTP server for ASP.NET Core. It runs directly on the `Http.Sys kernel driver <http://www.iis.net/learn/get-started/introduction-to-iis/introduction-to-iis-architecture>`_, and has very little overhead.
+Kestrel
+-------
 
-You can add support for WebListener to your ASP.NET application by adding the "Microsoft.AspNetCore.Server.WebListener" dependency in *project.json* and the following command:
+Kestrel is a cross-platform web server based on `libuv <https://github.com/libuv/libuv>`_, a cross-platform asynchronous I/O library. You add support for Kestrel by including ``Microsoft.AspNetCore.Server.Kestrel`` in your project's dependencies listed in *project.json* and calling :dn:method:`~Microsoft.AspNetCore.Hosting.WebHostBuilderKestrelExtensions.UseKestrel`.
 
-.. code-block:: javascript
+Learn more about working with Kestrel to create :doc:`/tutorials/your-first-mac-aspnet`.
 
-  "web": "Microsoft.AspNetCore.Hosting --server Microsoft.AspNetCore.Server.WebListener --server.urls http://localhost:5000"
+.. _weblistener:
 
-.. _kestrel:
+WebListener
+-----------
 
-Kestrel
--------
+WebListener is a Windows-only HTTP server for ASP.NET Core. It runs directly on the `Http.Sys kernel driver <http://www.iis.net/learn/get-started/introduction-to-iis/introduction-to-iis-architecture>`_, and has very little overhead.
```

Comment:
WebListener cannot be used with the ASP.NET Core Module, it can only be used independently.